pycaption/sami.py

```diff
@@ -71,21 +83,29 @@ def _get_xml_parser_class():
         """Hook method for providing a custom XML parser class"""
         return BeautifulSoup
 
-    def _build_layout(self, styles):
+    def _build_layout(self, styles, inherit_from=None):
         """
         :type styles: dict
-        :param styles: a dictionary CSS-like with styling rules
+        :param styles: a dictionary with CSS-like styling rules
+
+        :type inherit_from: Layout
+        :param inherit_from: The Layout with values to be used in case the
+            positioning settings in the styles parameter don't specify
+            something.
         """
         alignment = Alignment.from_horizontal_and_vertical_align(
             text_align=styles.get('text-align', None)
         )
-        layout = self._get_layout_class()(
+        new_layout = self._get_layout_class()(
```

Comment:
We talked about just creating new objects, to avoid proliferating deepcopy all over the place. Maybe use a parameter `inherit_from=layout` in the initializer, or a factory method Layout.inheriting_layout(layout) or something like that.